staticaddr/interface.go

```diff
@@ -0,0 +1,84 @@
+package staticaddr
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/btcsuite/btcd/btcec/v2"
+	"github.com/btcsuite/btcd/btcutil"
+	"github.com/lightningnetwork/lnd/keychain"
+)
+
+var (
+	ErrAddressAlreadyExists = fmt.Errorf("address already exists")
+	ErrAddressNotFound      = fmt.Errorf("address not found")
+)
+
+// AddressStore is the database interface that is used to store and retrieve
+// static addresses.
+type AddressStore interface {
+	// CreateStaticAddress inserts a new static address with its parameters
+	// into the store.
+	CreateStaticAddress(ctx context.Context,
+		addrParams *AddressParameters) error
+
+	// GetStaticAddress fetches static address parameters for a given
+	// address ID.
+	GetStaticAddress(ctx context.Context,
+		addressID ID) (*AddressParameters, error)
+
+	// DeleteStaticAddress deletes address parameters for a given static
+	// address ID.
+	DeleteStaticAddress(ctx context.Context,
+		addressID ID) error
+
+	// GetAllStaticAddresses retrieves all static addresses from the store.
+	GetAllStaticAddresses(ctx context.Context) (
+		[]*AddressParameters, error)
+}
+
+// ID is a unique identifier for a address.
+type ID [32]byte
+
+// FromByteSlice creates an address id from a byte slice.
+func (r *ID) FromByteSlice(b []byte) error {
+	if len(b) != 32 {
+		return fmt.Errorf("address id must be 32 bytes, got %d, %x",
+			len(b), b)
+	}
+
+	copy(r[:], b)
+	return nil
+}
+
+// AddressParameters holds all the necessary information for the 2-of-2 multisig
+// address.
+type AddressParameters struct {
+	// ID is a unique identifier for a static address
+	ID ID
+
+	// ClientPubkey is the client's pubkey for the static address. It is
+	// used for the 2-of-2 funding output as well as for the client's
+	// timeout path.
+	ClientPubkey *btcec.PublicKey
+
+	// ClientPubkey is the client's pubkey for the static address. It is
+	// used for the 2-of-2 funding output.
+	ServerPubkey *btcec.PublicKey
+
+	// Expiry is the CSV timout value at which the client can claim the
```

Comment:
I think you meant CLTV (so absolute time) right?

Reply:
CSV here is correct since the static address doesn't change, hence the expiry can't have absolute time since that would change the script hash every time.